src/metabase/task/search_index.clj

```diff
@@ -11,22 +11,32 @@
 
 (set! *warn-on-reflection* true)
 
-;; We need each instance to initialize on start-up currently, this will need to be refined.
-(jobs/defjob ^{DisallowConcurrentExecution false
+(def ^:private recreated? (atom false))
+
+(def job-key
+  "Key used to define and trigger the search-index task."
+  (jobs/key "metabase.task.search-index.job"))
+
+(jobs/defjob ^{DisallowConcurrentExecution true
                :doc                        "Populate Search Index"}
   SearchIndexing [_ctx]
-  (when (public-settings/experimental-fulltext-search-enabled)
-    (log/info "Recreating search index from latest schema")
-    (search/init-index! {:force-reset? true})
-    (log/info "Populating search index")
-    (search/reindex!)
+  (when (search/supports-index?)
+    (if (not @recreated?)
+      (do (log/info "Recreating search index from the latest schema")
+          ;; We need each instance to initialize on start-up currently, this will need to be refined.
```

Comment:
This is the comment, although its pretty vague.

One idea:

1. Update `init` to return false if the schema is already correct and there's nothing to do.
2. If it returned false, just re-index.

I think the "disable concurrent execution" flag sorts out locking for us, but otherwise we could take liquibase locks I guess.

Reply:
This still means that we wastefully re-index N - 1 times in a row for N instances starting starting. We could create a new table that keeps track of reindex metadata, like last time we did it, to debounce I guess.

Not that fluent in quartz design patterns 🤔

Reply:
Same here; I wonder if there is a way to make this work nicely and not over-complicate the code...

Reply:
Well, I want to do that graceful init stuff anyway? Restarting your instance shouldn't blast away your index.

Reply:
Updated the comment for now.